fuse_graphs/CMakeLists.txt

```diff
@@ -7,136 +7,93 @@ set(build_depends
   roscpp
 )
 
-find_package(catkin REQUIRED COMPONENTS
-  ${build_depends}
-)
-find_package(Boost REQUIRED)
+# Default to C++17
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CXX_STANDARD_REQUIRED YES)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+find_package(ament_cmake REQUIRED)
+find_package(fuse_core REQUIRED)
+find_package(pluginlib REQUIRED)
+find_package(rclcpp REQUIRED)
+
 find_package(Ceres REQUIRED)
 
-catkin_package(
-  INCLUDE_DIRS
-    include
-  LIBRARIES
-    ${PROJECT_NAME}
-  CATKIN_DEPENDS
-    ${build_depends}
-  DEPENDS
-    Boost
-    CERES
-)
+include(boost-extras.cmake)
 
 ###########
 ## Build ##
 ###########
-add_compile_options(-Wall -Werror)
 
 ## fuse_graphs library
-add_library(${PROJECT_NAME}
+add_library(${PROJECT_NAME} SHARED
```

Comment:
Same comment about leaving off `SHARED` and using `ament_cmake_ros` https://github.com/locusrobotics/fuse/pull/287#discussion_r1028631887